src/test/java/org/jabref/migrations/SpecialFieldsToSeparateFieldsTest.java

```diff
@@ -0,0 +1,86 @@
+package org.jabref.migrations;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.SpecialField;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class SpecialFieldsToSeparateFieldsTest {
+
+    @ParameterizedTest
+    @MethodSource("provideKeywordFieldPairs")
+    public void migrateToCorrectField(SpecialField expectedField, String fieldInKeyword) {
+        BibEntry entry = new BibEntry().withField(StandardField.AUTHOR, "JabRef")
+                .withField(StandardField.KEYWORDS, fieldInKeyword);
+
+        new SpecialFieldsToSeparateFields(',').performMigration(new ParserResult(List.of(entry)));
+
+        assertEquals(fieldInKeyword, entry.getField(expectedField).get());
+    }
+
+    @Test
+    public void noKewordToMigrate() {
+        BibEntry entry = new BibEntry().withField(StandardField.AUTHOR, "JabRef")
+                                       .withField(StandardField.KEYWORDS, "tdd");
+        new SpecialFieldsToSeparateFields(',').performMigration(new ParserResult(List.of(entry)));
+
+        assertEquals("JabRef", entry.getField(StandardField.AUTHOR).get());
+        assertEquals("tdd", entry.getField(StandardField.KEYWORDS).get());
+    }
+
+    @Test
+    public void migrateMultipleSpecialFields() {
+        BibEntry entry = new BibEntry().withField(StandardField.AUTHOR, "JabRef")
+                                       .withField(StandardField.KEYWORDS, "printed, prio1");
+        new SpecialFieldsToSeparateFields(',').performMigration(new ParserResult(List.of(entry)));
+
+        assertEquals("prio1", entry.getField(SpecialField.PRIORITY).get());
+        assertEquals("printed", entry.getField(SpecialField.PRINTED).get());
+
```

Comment:
It should be checked, whether the keyword field has `printed` and `prio1` removed.

Why not checking an expected BibEntry? --> Applies for all test cases 😇